### ISSUE TITLE
[ESP32/M5 board] Pairing All clusters app with non default Discriminator & setup pin code fails with PASE Errors

### ISSUE THREAD

**Issue by kean-apple** [project-chip/connectedhomeip#18037]
#### Problem
Pairing with non default Discriminator & setup pin code fails with PASE Errors. This happens on the M5 board All clusters app. Did not issue this issue when running all clusters app on Mac

SHA: cfc35951be66a664a6efdadea56d1b8ea6e63e96 (and TE8 SHA)

Steps:
1. In CHIPDeviceConfig.h: set the following 
    ```
    #define CHIP_DEVICE_CONFIG_USE_TEST_SETUP_DISCRIMINATOR 0xF07
    #define CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE 20202027
    ```
2. Build M5 All clusters app
3. Using CHIPTool CLI or iOS CHIPtool, scan the QR code to start pairing (QR code contents looks updated with discriminator & setup pin above)
4. Pairing fails due to PASE error
```
2022-05-03 19:47:17.058599-0700  localhost CHIPTool[10741]: (CHIP) [com.zigbee.chip:all] ð´ [1651632437058] [10741:866985] CHIP: [SC] Failed during PASE session setup. ../../../../../../../../../../../connectedhomeip/src/crypto/CHIPCryptoPALmbedTLS.cpp:1078: CHIP Error 0x000000AC: Internal error
2022-05-03 19:47:17.058638-0700  localhost CHIPTool[10741]: (CHIP) DevicePairingDelegate status updated: 1
```
[PASE-Session-failed-M5-TE9-logs.txt](https://github.com/project-chip/connectedhomeip/files/8620702/PASE-Session-failed-M5-TE9-logs.txt)
[PASE-session-CHIP-tool-TE9-logs.txt](https://github.com/project-chip/connectedhomeip/files/8620703/PASE-session-CHIP-tool-TE9-logs.txt)
![IMG_0115](https://user-images.githubusercontent.com/73258860/166690651-609813f3-3828-48b7-a3bc-131e68431d33.PNG)

Note: Pairing works if using default discriminator & setup code.

**Comment by bzbarsky-apple** [comment]
OK.  So on the assumption that what was changed is `src/include/platform/CHIPDeviceConfig.h`, here's what I see:

1. The setup codes the m5stack claims via its logging say the setup PIN is 20202027.
2. Trying to commission using 20202027 as the setup PIN fails.
3. Trying to commission using 20202021 as the setup PIN works.

With all-clusters-app on Mac, if I make that same change I see the following:

1. The setup codes the Mac all-clusters-app claims via its logging say the setup PIN is 20202021.
2. Trying to commission using 20202027 as the setup PIN fails.
3. Trying to commission using 20202021 as the setup PIN works.

So in both cases the device still needs the setup PIN to be 20202021; the difference is what they advertise in their pairing codes.

Comparing how the two applications work:

The "linux" (and Mac) version of all-clusters-app uses `LinuxCommissionableDataProvider` and calls `SetCommissionableDataProvider` in  `ChipLinuxAppInit` before printing the onboarding codes.  This provider is initialized via `InitCommissionableDataProvider` which will use command-line options and fall back to `TestOnlyCommissionableDataProvider` if those are not provided.  `TestOnlyCommissionableDataProvider` hardcodes 20202021 and 3840 for the PIN code and discriminator, and ignores `CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE` and `CHIP_DEVICE_CONFIG_USE_TEST_SETUP_DISCRIMINATOR`.

The esp32 version of all-clusters-app does not call `SetCommissionableDataProvider` on its own and as `chip_use_transitional_commissionable_data_provider` defined.  So it ends up using a `LegacyTemporaryCommissionableDataProvider` , set up in `GenericConfigurationManagerImpl::Init`.  This will use `CHIP_DEVICE_CONFIG_USE_TEST_SETUP_DISCRIMINATOR` and `CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE`.  But it will also use `CHIP_DEVICE_CONFIG_USE_TEST_SPAKE2P_VERIFIER`, which needs to match the setup PIN code.  And per the steps above, it does not look like the verifier was modified, so the actual PIN code which will make the crypto work is unchanged; it's just all the places that look at the PIN code directly that got changed.

Long story short:

1. Changing  these `CHIP_DEVICE_CONFI_USE_TEST_*`  values is not a reliable way to change the example app behavior; whether it works or not depends on the platform.
2. Changing just the PIN without changing the verifier will not work.
